docs/manual/guides/form-save.de.md

```diff
@@ -0,0 +1,238 @@
+Wir benötigen Einträge für die Felder »pid«, »title«, »alias«, »author«, »startDate«, »startTime«, »endDate«, 
+»endTime« und »published«, damit wir im Backend eine gültige Darstellung unserer Reservierungen erhalten. Optional 
+wären auch Einträge für »location«, »description« oder »teaser« möglich. 
+
+Unser Formular bedient zur Zeit zumindest die Felder »title«, »startDate« und optionales »endDate«. Die weiteren Felder
+könnte man über versteckte Formularfelder hinzufügen. Diese Vorgehensweise ermöglicht jedoch die Manipulation der zu 
+übermittelten Formulardaten. Aus diesem Grund verwenden wir den Hook 
+»[prepareFormData](https://docs.contao.org/dev/reference/hooks/prepareFormData/)«.
+
+
+Dieser Hook greift nach der Formular-Übermittlung aber vor der eigentlichen Datenspeicherung. Du kannst darüber 
+die übermittelten Daten bearbeiten und neue Informationen hinzufügen. Erstelle in deiner Contao-Installation 
+die Verzeichnisse ``src\EventListener\`` und platziere darin die Datei ``PrepareFormDataListener.php`` mit folgendem Inhalt:
+
+
+```php
+// src/EventListener/PrepareFormDataListener.php
+
+namespace App\EventListener;
+
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\Form;
+
+use Contao\CoreBundle\Slug\Slug;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @Hook("prepareFormData")
+ */
+class PrepareFormDataListener 
+{
+    private $slug;
+    private $db;
+
+    public function __construct(Slug $slug, Connection $db)
+    {
+        $this->slug = $slug;
+        $this->db = $db;
+    }  
+  
+    public function __invoke(array &$submittedData, array $labels, array $fields, Form $form): void
+    {
+        // set id of form, eventarchiv and author
+        $idForm         = 3;
+        $idEventArchiv  = 5;
+        $idAuthor       = 1;
+    
+        // form restriction 
+        if ((int) $form->id === $idForm) {
+
+          // mandatory fields
+          $submittedData['pid']       = $idEventArchiv;
+          $submittedData['author']    = $idAuthor;
+          $submittedData['published'] = 1;
+        
+          // generate unique alias 
+          $submittedData['alias'] = $this->getSlug($submittedData['title']);
+      
+          $submittedData['startTime'] = strtotime($submittedData['startDate']);
+        
+          // optional fields
+          if (!empty(trim($submittedData['endDate']))) {
+            $submittedData['endTime'] = strtotime($submittedData['endDate']);
+          } else {
+            $submittedData['endTime'] = null;
+            $submittedData['endDate'] = null;
+          }
+        }
+    }
+  
+    public function getSlug(string $text, string $locale = 'de', string $validChars = '0-9a-z'): string
+    {
+        $options = [
+          'locale' => $locale,
+          'validChars' => $validChars,
+        ];
+
+        $duplicateCheck = function (string $slug): bool {
+          return $this->slugExists($slug);
+        };
```

Comment:
Please fix the indentations ;)

Reply:
?? Seems OK Or what do you mean

Reply:
You are mixing 2 and 4 spaces.

Reply:
Yepp, but this will often produce unnessary line breaks within the Documentation. 
?

Reply:
That's ok. You can also try to decrease the indentation level with early-outs. For example, instead of writing

```php
// form restriction 
if ((int) $form->id === $idForm) {
    // …
}
```

you could write

```php
if ((int) $form->id !== $idForm) {
    return;
}

// …
```